kornia/augmentation/container/augmented_tensor.py

```diff
@@ -0,0 +1,101 @@
+from typing import List, Sequence, Union, cast
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+from torch._C import dtype
+
+
+def pad_batch_list(data: Sequence[Tensor], padding_value=0.0):
+    batch_dim = len(data)
+    if batch_dim > 0:
+        if isinstance(data, list):
+
+            def calc_dim_pad(stat):
+                return tuple(int(item) for size in stat.flip(dims=(0,)) for item in (0, size))
+
+            dims = torch.IntTensor([d.shape for d in data])
+            max_dims = torch.amax(dims, dim=0)
+            padding_dims = torch.stack([max_dims - torch.IntTensor(list(d.shape)) for d in data])
+            padded = torch.stack(
+                [F.pad(d, calc_dim_pad(padding_dims[i]), value=padding_value) for i, d in enumerate(data)]
+            )
+            return padded, padding_dims
+    return data, torch.empty(0)
+
+
+class AugmentedTensor(torch.Tensor):
```

Comment:
thinking that this could be the starting of a `kornia.core` submodule under the name of `ImageTensor` or `Image` to match with https://github.com/kornia/kornia_io/blob/master/kornia_io/image.py#L20

/cc @shijianjian @ducha-aiki @lferraz @anguelos @Borda

Reply:
@shijianjian and myself are working on this, it is still a draft. we thought about wrapping all internal tensors.